File: frontend/src/services/comment.service.ts
import { http } from './request';
import { Comment } from '@/types';

export interface CommentQueryDTO {
  current?: number;
  size?: number;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
}

export interface PageResult<T> {
  records: T[];
  total: number;
  current: number;
  size: number;
}

/**
 * 评论服务
 *
 * <p>后端接口路径：/api/comments</p>
 * <p>注意：评论接口独立于文档接口，不在 /documents 路径下</p>
 */
export const commentService = {
  // 创建评论
  createComment: (data: {
    documentId: string | number;
    content: string;
    parentId?: string | number;
  }) => {
    return http.post<number>('/document/comments', data);
  },

  // 删除评论
  deleteComment: (commentId: string | number) => {
    return http.delete(`/document/comments/${commentId}`);
  },

  // 点赞评论
  likeComment: (commentId: string | number) => {
    return http.post(`/document/comments/${commentId}/like`);
  },

  // 取消点赞评论
  unlikeComment: (commentId: string | number) => {
    return http.delete(`/document/comments/${commentId}/like`);
  },

  // 分页查询文档评论（注意：后端使用POST请求）
  pageDocumentComments: (
    documentId: string | number,
    query: CommentQueryDTO = {}
  ) => {
    const params = {
      current: query.current || 1,
      size: query.size || 10,
      sortBy: query.sortBy || 'createdAt',
      sortOrder: query.sortOrder || 'desc',
    };
    return http.post<PageResult<Comment>>(
      `/document/comments/document/${documentId}`,
      params
    );
  },

  // 获取评论回复列表
  getCommentReplies: (parentCommentId: string | number) => {
    return http.get<Comment[]>(`/document/comments/${parentCommentId}/replies`);
  },
};

export default commentService;

File: backend/kb-core/kb-core-document/src/main/java/com/knowledge/base/document/service/CommentService.java
package com.knowledge.base.document.service;

import com.baomidou.mybatisplus.extension.service.IService;
import com.knowledge.base.common.result.PageResult;
import com.knowledge.base.document.dto.CommentCreateDTO;
import com.knowledge.base.document.dto.CommentQueryDTO;
import com.knowledge.base.document.entity.Comment;
import com.knowledge.base.document.vo.CommentVO;

import java.util.List;

/**
 * 评论Service接口
 *
 * @author 苏三
 * @since 1.0.0
 */
public interface CommentService extends IService<Comment> {

    /**
     * 创建评论
     *
     * @param dto 创建DTO
     * @return 评论ID
     */
    Long createComment(CommentCreateDTO dto);

    /**
     * 删除评论
     *
     * @param commentId 评论ID
     * @return 是否成功
     */
    Boolean deleteComment(Long commentId);

    /**
     * 点赞评论
     *
     * @param commentId 评论ID
     * @return 是否成功
     */
    Boolean likeComment(Long commentId);

    /**
     * 取消点赞评论
     *
     * @param commentId 评论ID
     * @return 是否成功
     */
    Boolean unlikeComment(Long commentId);

    /**
     * 分页查询文档评论
     *
     * @param documentId 文档ID
     * @param dto 查询DTO
     * @return 分页结果
     */
    PageResult<CommentVO> pageDocumentComments(Long documentId, CommentQueryDTO dto);

    /**
     * 获取评论回复列表
     *
     * @param parentCommentId 父评论ID
     * @return 回复列表
     */
    List<CommentVO> getCommentReplies(Long parentCommentId);
}
